kubernetes/provider.go

```diff
@@ -212,6 +264,10 @@ func (k kubeClientsets) MainClientset() (*kubernetes.Clientset, error) {
 		return k.mainClientset, nil
 	}
 
+	if err := checkConfigurationValid(k.configData); err != nil {
+		return nil, err
+	}
+
```

Comment:
I did a bunch of Progressive Apply testing and I wasn't able to find a difference between having this check enabled vs disabled.